### ISSUE TITLE
buffer previewer: keep_last_buf also happens on escape

### ISSUE THREAD

**Issue by Conni2461** [nvim-telescope/telescope.nvim#575]
I currently document previewers and i just realized something.
When using `keep_last_buf` i need to clear the buffer when i close the picker without selecting something. This currently doesn't happen which makes this feature, potentially leak always one buffer.

This feature is not used in core but `octo` and `cheat` are using it. And i suggested it for `arecibo`.

I am thinking for the last 5 minutes how to solve that issues but i could come up with a solution because the previewer doesn't know the calling action. I need to think about this again. Maybe my implementation isn't perfect.

**Comment by matu3ba** [comment]
![exiting_neovim](https://user-images.githubusercontent.com/13089667/110021983-b3e30f00-7d2b-11eb-935a-9ed7e6471ebf.png)

Like this?

**Comment by Conni2461** [comment]
No. What did you do what did you ran? This shows a scrolling error

**Comment by matu3ba** [comment]
I wanted to exit neovim and ran `:q` repeatedly. Probably `:qall` would have been more optimal.

**Comment by Conni2461** [comment]
You probably did run into this issue at some point and the prompt buffer wasn't cleaned up. Happens really often when you develop for telescope and mess around with stuff. Haven't seen it in production tho. If that happens `:qa!` because the prompt buffer still has some content. But if you haven't used `octo` this issue is not related. `keep_last_buf` is not used in core. If that happens again and you have a somewhat reproduce i would appreciate a bug report :)
